lib/Support/Pipeline.cpp

```diff
@@ -123,6 +123,7 @@ void MappingTraits<offloadtest::Resource>::mapping(IO &I,
   I.mapRequired("Name", R.Name);
   I.mapRequired("Kind", R.Kind);
   I.mapRequired("DirectXBinding", R.DXBinding);
+  I.mapRequired("VulkanBinding", R.VKBinding);
```

Comment:
I kinda feel like it was a mistake to make the DirectXBinding required, since we only need it if you don't have an embedded root signature.

Should we make the Vulkan binding required or optional? I think there are a few good arguments for optional, but curious about your thoughts?

Reply:
I mostly made it required to match the DirectX binding, and don't have a strong opinion. It is odd that we need to specify this in places like the root signature tests where it isn't actually used. However, making it required definitely simplifies some things - if we don't have a vulkan binding, what should we do? Assuming the bindings increase in source order like this used to do tends to fall apart and lead to rather confusing error conditions where we write to the wrong buffers and such.

Reply:
I've updated this to make the vulkan binding optional, but we'll error if we try to run on vulkan and it's missing. I think this is a good middle ground and we should consider something similar for the DX binding.